src/agent_workflow_server/services/runs.py

```diff
@@ -172,6 +173,26 @@ def search(search_request: RunSearchRequest) -> List[ApiRun]:
             islice(islice(runs, search_request.offset, None), search_request.limit)
         )
 
+    @staticmethod
+    async def resume(run_id: str, user_input: Dict[str, Any]) -> ApiRun:
+        run = DB.get_run(run_id)
+        if run is None:
+            raise ValueError("Run not found")
+        if run["status"] != "interrupted":
+            raise ValueError("Run is not in interrupted state")
+        if run["interrupt"] is None:
```

Comment:
"interrupt" is Optional in models.py, it mean that it can missing. I understand that if run["status"]=="interrupted" means you must have a "interrupt" key.
Let as this if you are sure, otherwise prefer:
```
if "interrupt" not in run or run["interrupt"] is None:
   ...
```
Or use a simple run.get("interrupt", None)